src/doghouse/core/domain/snapshot.py

```
@@ -0,0 +1,70 @@
+from dataclasses import dataclass, field, asdict
+from typing import List, Dict, Any, Optional
+from .blocker import Blocker, BlockerType, BlockerSeverity
+
+@dataclass(frozen=True)
+class Snapshot:
+    timestamp: datetime.datetime
+    head_sha: str
+    blockers: List[Blocker]
+    metadata: Dict[str, Any] = field(default_factory=dict)
+
+    def __post_init__(self):
```

Ruff (0.15.7)

[warning] Missing return type annotation for special method `__post_init__`

(ANN204)